packages/node/src/plugins/segmentio/publisher.ts

```diff
@@ -58,21 +61,31 @@ export class Publisher {
       httpRequestTimeout,
       httpClient,
       disable,
+      oauthSettings: oauthSettings,
     }: PublisherProps,
     emitter: NodeEmitter
   ) {
     this._emitter = emitter
     this._maxRetries = maxRetries
     this._maxEventsInBatch = Math.max(maxEventsInBatch, 1)
     this._flushInterval = flushInterval
-    this._auth = b64encode(`${writeKey}:`)
     this._url = tryCreateFormattedUrl(
       host ?? 'https://api.segment.io',
       path ?? '/v1/batch'
     )
     this._httpRequestTimeout = httpRequestTimeout ?? 10000
     this._disable = Boolean(disable)
     this._httpClient = httpClient
+    this._writeKey = writeKey
+
+    if (oauthSettings) {
+      const TokenManager = dependencyInjection.get('TokenManager')
+      oauthSettings.httpClient ??= httpClient
+      oauthSettings.maxRetries ??= maxRetries
+      this._tokenManager = new TokenManager({
+        ...oauthSettings,
```

Comment:
Do we actually want to copy? Are we mutating? a bit hard to grasp the intention, and we probably don't have tests for this.
 
This is a bit confusing, because like 84 is mutating the settings and storing them, but then we are passing a copy on line 86.

Reply:
Nope, it was left over from trying to jam the setting of httpClient and maxRetries into one line with instantiating the token manager and I didn't revert it after moving them above.
I feel like this is publisher modifying the behavior of the token manager class for its own purposes, which doesn't preclude token manager from having its own defaults.